pilot/pkg/xds/eds.go

```diff
@@ -434,6 +434,20 @@ func canSendPartialFullPushes(req *model.PushRequest) bool {
 	return true
 }
 
+func canUseEdsCache(req *model.PushRequest) bool {
```

Comment:
I will note this will reduce cache hits.

Before, when we get len==0 or PA change, we clear the cache

But then we generate endpoints for all proxies. So after the first proxy is done, cache is populated and all remaining proxies can use it. Now we cannot do that.

I think we can actually remove the len=0 check since we already Clear() the whole cache in that case.

Then we are just making a tradeoff on the cost of DependantTypes VS the cost of recomputing all on PA change

Reply:
I can think of a way to resolve it.